modules/frontend/search_progress.go

```diff
@@ -185,6 +193,13 @@ func (r *searchProgress) result() *shardedSearchResults {
 	return res
 }
 
+func (r *searchProgress) metrics() *tempopb.SearchMetrics {
+	r.mtx.Lock()
+	defer r.mtx.Unlock()
+
+	return r.resultsMetrics
```

Comment:
this will create race conditions on the metric fields b/c you are returning a pointer to a memory location that other goroutines have access to and will attempt to update. 

you should likely clone this, but that can be costly. if you call it occassionally it should be ok.

Reply:
updated this to return a copy resultsMetrics.

this is only called once each tenant (when we create a progress project), so should be fine.